python/cudf/dataframe/numerical.py

```diff
@@ -216,8 +207,8 @@ def unique(self, method='sort'):
             raise NotImplementedError(msg)
         segs, sortedvals = self._unique_segments()
         # gather result
-        out = cudautils.gather(data=sortedvals, index=segs)
-        return self.replace(data=Buffer(out), mask=None)
+        out_col = cpp_copying.apply_gather_array(sortedvals, segs)
```

Comment:
Hmmm, so `unique()` calls `_unique_segments()` which calls `sort_by_values` which calls `apply_gather`, and then `unique()` calls `apply_gather` again?  (This all needs to move to libcudf...)